accepted/2025/nuget-mcp.md

```diff
@@ -0,0 +1,265 @@
+
+#### Startup instructions
+
+We have two options to encode the startup instructions into the package:
+
+1. Instruct the package author to include the desired consumer MCP JSON in the readme, and allow NuGet.org to scrape the JSON from the README markdown (a code block matching a certain pattern).
+2. Embed standardized, machine-readable information in the package to allow tooling to know of the startup instructions required for the MCP server
+   - This could be the same [author `mcp.json`](https://github.com/modelcontextprotocol/registry/blob/main/tools/publisher/README.md) used for publishing to the MCP registry, embedded in the root of the package.
+   - This could be information in a new .nuspec field.
+   - The core idea here is to include similar information as the [`package` entity in the MCP registry Open API schema](https://github.com/modelcontextprotocol/registry/blob/3df06d38d9b6590f6ba9bfde56fb3583d8ff4e9d/docs/openapi.yaml#L165-L200). This is essentially the information provided by the MCP registry to MCP client tooling to enable MCP server installation and setup.
+   - See [modelcontextprotocol/registry#118](https://github.com/modelcontextprotocol/registry/discussions/118) for discussion.
+
+The JSON we would like to either scrape from the README (option 1 above) or generate from machine-readable startup instructions (options 2 above) would look like this.
+
+```json
+{
+  "inputs": [
+    {
+      "type": "promptString",
+      "id": "contoso-api-key",
+      "description": "Contoso API Key",
+      "password": true
+    }
+  ],
+  "servers": {
+    "Contoso.MySuperMcp": {
+      "type": "stdio",
+      "command": "dnx",
+      "args": ["Contoso.MySuperMcp", "--", "mcp", "start"],
+      "env": {
+        "CONTOSO_API_KEY": "${input:contoso-api-key}"
+      }
+    }
+  }
+}
+```
+
+For option 1, NuGet.org would look in README.md for a JSON code block with a `servers` JSON property containing a property matching the current package ID. If found, it will place the JSON in the command palette for easy copying. Other MCP JSON configuration shapes will be investigated also. It appears Anthropic uses `mcpServers` in their JSON. If no recognized JSON format is found, a default MCP JSON will be generated:
+
+
+```json
+{
+  "servers": {
+    "Contoso.MySuperMcp": {
+      "type": "stdio",
+      "command": "dnx",
+      "args": ["Contoso.MySuperMcp", "--version", "1.0.1"]
+    }
+  }
+}
+```
+
+#### Project template
+
+To improve the project setup experience, we will introduce a new project template. The template will be available via `dotnet new mcp-server`. This will create a .csproj for a CLI tool, with a stable MCP SDK dependency version, and an `McpServer` server package type.
+
+The .csproj will have the following shape:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net10.0</TargetFramework>
+    <RootNamespace>Contoso.MySuperMcp</RootNamespace>
```

Comment:
`RootNamespace` isn't customary for a minimal SDK style project these days, is it?